js/output/sql/sql-output.js

```diff
@@ -122,6 +122,26 @@ window.SQLOutput = Backbone.View.extend({
             errorMessage = i18n._("There's a syntax error near %(nearThing)s.",
                 {nearThing: nearPhrase.substr(5)});
         }
+        const colValueMismatchReg = /has (\d+) columns but (\d+) values were supplied/;
+        const colMismatchError = sqliteError.match(colValueMismatchReg);
+        if (colMismatchError) {
+          const tabName = sqliteError.split(" ")[1];
+          const numCols = colMismatchError[1];
+          const numVals = colMismatchError[2];
+
+          const tableHasColumns = i18n.ngettext(
+              "Table \"%(tableName)s\" has 1 column",
+              "Table \"%(tableName)s\" has %(num)s columns",
+              numCols, {tableName: tabName});
+
+          const valuesSupplied = i18n.ngettext(
+              "but 1 value was supplied.",
+              "but %(num)s values were supplied.",
+              numVals);
+
+          errorMessage = i18n._("%(columns)s %(values)s",
```

Comment:
Is there a way to add context for this strings so translators are aware what this referes to?

Reply:
This is how we do it in the main codebase:
// I18N: "bold" as in bolded text that's thicker for emphasis.
bold: i18n._("bold"),

I think it might work here? It's worth a shot.

Reply:
Yes, `// I18N: Explanation here` works in JS too. In `ngettext`, the third parameter is used to select the plurality at runtime, and also automatically becomes the named parameter `%(num)s`, so it's correct that the parameter dict in the first call only needs to contain `tableName` and that the second call doesn't need a parameter dict. However, 1 must not be hardcoded in the singular string, because then the linter won't let the translation contain `%(num)s`, and some languages use the "singular case" for other numbers than 1.

Reply:
@aasmundeldhuset thank you for confirming and for spotting an issue. What a silly mistake for me to make, since I am well aware of this problem. Not sure what I was thinking. :thinking: Will fix and add a comment soon.